### DISCUSSION TITLE
AMD - Windows - What are our options?

### DISCUSSION THREAD

**Discussion by ryan-reddekopp** [vladmandic/sdnext#573] [General]
I saw that a thread was recently closed discussing AMD, and there was lots of great conversation happening.
My question is that ROCm was the primary topic which makes me think that development was mostly focused on Linux.

Is our only option for AMD/Windows the DirectML branch by lshqqytiger https://github.com/vladmandic/automatic/tree/directml, or does the ROCm implementation work with Windows?

There was also doggettx being thrown around, but I am unfamiliar with that tech.

Anyway, I guess i was just looking for clarification. Thanks!

**Comment by vladmandic** [comment]
DirecML branch is where development is happening, when it's ready, it will be merged. In the meantime, you're welcome to test it.
Regarding ROCm, AMD said that next version should support Windows - if it does, we'll do our best.

**Reply by ryan-reddekopp** [reply]
Thanks so much for the feedback! All your hard work is much appreciated.

**Reply by iDeNoh** [reply]
I'm excited to see how well ROCM works on windows, DirectML is... Fine, but it's nowhere near as good as rocm in terms of performance.

**Reply by brknsoul** [reply]
If so, that'll be rad! No more dual-booting!

**Reply by vladmandic** [reply]
@brknsoul did you try WSL2 under Windows? I've been using it ever since GPU support was added to it and its my primary environment.

**Reply by motorist828** [reply]
>  did you try WSL2 under Windows?

ROCM does not work in WSL as far as I know
I have tried this before but didn't get any result.
perhaps something has changed since then

**Reply by vladmandic** [reply]
@motorist828 unfortunately, it seems thats the case - and even more unfortunate is that it would be easy for AMD to support it since everything is already ready & baked in. see <https://github.com/RadeonOpenCompute/ROCm/issues/794>

**Reply by brknsoul** [reply]
I have your fork of SD running in Ubuntu 22. It was surprisingly easy to install. The hard part was getting ROCm to install, but after that, I just cloned your repo, linked some folders from lshqqytiger's fork, and then ran webui.sh.

On my AMD RX 6600 XT 8Gb, I get around 4-4.5 it/s, on a Euler a / 20 steps / 512x512 image.

I went from 1 min per image in Windows, to 4 sec per image in Ubuntu!

**Reply by iDeNoh** [reply]
Were there any changes in the last day that would affect AMD performance? I had a few days where I couldn't be at my computer much and before I was getting approximately around 6.2it/s with the version I was on prior to this (I don't recall which unfortunately) and when I updated to the latest commit I'm not getting 7.6it/s without changing anything other than the commit version.
